flocker/common/functional/test_script.py

```diff
@@ -172,3 +184,35 @@ def test_sigint(self):
                                 u"message": u"Received SIGINT, shutting down.",
                                 u"error": False}))
         return d
+
+    def _assert_logfile_messages(self, stdout_messages, logfile):
+        """
+        Verify that the messages have been logged to a file rather than to
+        stdout.
+        """
+        self.assertEqual([], stdout_messages)
+        logfile_messages = map(loads, logfile.getContent().splitlines())
+        assertContainsFields(
+            # message[0] contains a twisted log message.
+            self, logfile_messages[1], {u"key": 123}
+        )
+
+    def test_file_logging(self):
+        """
+        Logged messages are written to ``logfile`` if ``--logfile`` is supplied
+        on the command line.
+        """
+        logfile = FilePath(self.mktemp()).child('foo.log')
+        logfile.parent().makedirs()
```

Comment:
:question: doesn't this throw an `OSError` if the directory matching `logfile.parent()` already exists?